executor/executor_test.go

```diff
@@ -2787,3 +2787,57 @@ func (s *testSuite) TestYearTypeDeleteIndex(c *C) {
 	tk.MustExec("delete from t;")
 	tk.MustExec("admin check table t")
 }
+
+func (s *testSuite) TestUnsignedDecimalOverflow(c *C) {
+	tests := []struct {
+		input  interface{}
+		hasErr bool
+		err    string
+	}{{
+		-1,
+		true,
+		"Out of range value for column",
+	}, {
+		"-1.1e-1",
+		true,
+		"Out of range value for column",
+	}, {
+		-1.1,
+		true,
+		"Out of range value for column",
+	}, {
+		-0,
+		false,
+		"",
+	},
+	}
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t(a decimal(10,2) unsigned)")
+	for _, t := range tests {
+		func() {
```

Comment:
Why wrap in a function?

Reply:
I want to use defer to close result, so for each loop make a function- -
yes, for test suit is not nessary and I removed that.